ui/algorithm.h

```diff
@@ -5,13 +5,17 @@
 #ifndef AMOEBOTSIM_UI_ALGORITHM_H_
 #define AMOEBOTSIM_UI_ALGORITHM_H_
 
+#include <memory>
 #include <utility>
 #include <vector>
 
 #include <QString>
 #include <QStringList>
+#include <QObject>
 
-class Algorithm {
+#include <core/system.h>
+
+class Algorithm : public QObject {
  public:
```

Comment:
You're missing a `Q_OBJECT` guard right here. See, for example, `core/simulator.h`.

Reply:
That will at least get the thing to build, but you'll probably need to do a lot more testing to make sure everything actually works.